src/db/migrate.rs

```diff
@@ -322,6 +323,32 @@ fn migrate_inner(
         ),
     ];
 
+    if let ApplyMode::Permanent = apply_mode {
+        migrations.push(migration!(
```

Comment:
With the current migration setup running the tests will attempt to recreate this function and index on each test. There's probably some nicer way to handle this than just not registering the migration during testing, I'm open to suggestions.

Reply:
I'm confused why you don't want to register the function during testing? Isn't that part of what we want to test?

Reply:
It attempts to create the function multiple times, failing because it already exists. I guess the issue with this setup now is that if you try and run the tests without migrating the database manually the function won't exist at all.

Reply:
Yes, I would like tests to be as automated as possible. Maybe you could use `CREATE OR REPLACE FUNCTION` instead?

Reply:
Changed the test setup to create a separate schema for each test run, so they're running on nearly fully independent databases.